openstf/model/objective_creator.py

```diff
@@ -13,6 +13,14 @@ class ObjectiveCreator:
 
     @staticmethod
     def create_objective(model_type: Union[MLModelType, str]) -> RegressorObjective:
+        if model_type not in ObjectiveCreator.OBJECTIVES.keys():
+            raise NotImplementedError(
+                "This model_type is not implemented "
+                "for hyperparam optimization."
+                f"Received: {model_type}, "
+                "Should be in: "
+                f"{list(ObjectiveCreator.OBJECTIVES.keys())}"
```

Comment:
Perhaps to make the message a little shorter and more readable you could use:

```python
        valid_types = list(ObjectiveCreator.OBJECTIVES.keys())
        if model_type not in valid_types:
            raise NotImplementedError(
                "This model_type is not implemented "
                "for hyperparam optimization."
                f"Received: {model_type}, "
                "Should be in: "
                f"{', '.join([t.value for t in valid_types])}"
            )
```

This gives `"xgb"`, while the current implementation gives `[<MLModelType.XGB: 'xgb'>]`.

Minor remark. This module does not need knowledge about hyperparameters optimization. It just provides an objective function. So you could be more precise with the error message on what's going on; the objective function for this model type is not implemented yet: `f"No objective function for {model_type} valid model_types are: {', '.join([t.value for t in valid_types])}"`

Reply:
Thanks for your suggestion, I'll incorporate it